### PR TITLE
[CodeCompletion] Remove warning for 'async in non-concurrency context'

### PR DESCRIPTION
**Pull request by rintaro**
This warnings don't give much benefits for developers. Code completion UI tends to show them unusable. But usually, developers can modify the context to accept async calls, e.g. by wrapping it with `Task { }`

rdar://126737530

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/CodeCompletionResult.cpp [hunk 1]
```diff
@@ -399,9 +399,6 @@ ContextFreeCodeCompletionResult::calculateContextualNotRecommendedReason(
   if (explicitReason != ContextualNotRecommendedReason::None) {
     return explicitReason;
   }
-  if (IsAsync && !canCurrDeclContextHandleAsync) {
-    return ContextualNotRecommendedReason::InvalidAsyncContext;
-  }
```
Comment (ahoppen) [inline lib/IDE/CodeCompletionResult.cpp:404]: Unless I’m missing something, we can remove `IsAsync` from `ContextFreeCodeCompletionResult` now.

**Review by ahoppen — APPROVED** [review]

File: lib/IDE/CodeCompletionDiagnostics.cpp [hunk 2]
```diff
@@ -166,10 +166,6 @@ bool swift::ide::getContextualCompletionDiagnostics(
     const ASTContext &Ctx) {
   CodeCompletionDiagnostics Diag(Ctx);
   switch (Reason) {
-  case ContextualNotRecommendedReason::InvalidAsyncContext:
-    // FIXME: Could we use 'diag::async_in_nonasync_function'?
-    return Diag.getDiagnostics(
-        Severity, Out, diag::ide_async_in_nonasync_context, NameForDiagnostics);
```
Comment (hamishknight) [inline lib/IDE/CodeCompletionDiagnostics.cpp:172]: Is `diag::ide_async_in_nonasync_context` now unused?

**Review by hamishknight — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/CodeCompletionCache.cpp [hunk 3]
```diff
@@ -267,8 +266,8 @@ static bool readCachedModule(llvm::MemoryBuffer *in,
 
     ContextFreeCodeCompletionResult *result =
         new (*V.Allocator) ContextFreeCodeCompletionResult(
-            kind, associatedKind, opKind, roles, isSystem, isAsync,
-            hasAsyncAlternative, string, moduleName, briefDocComment,
+            kind, associatedKind, opKind, roles, isSystem,
+                                                           hasAsyncAlternative, string, moduleName, briefDocComment,
```
Comment (ahoppen) [inline lib/IDE/CodeCompletionCache.cpp:270]: Incorrect formatting?

File: lib/IDE/CodeCompletionCache.cpp [hunk 4]
```diff
@@ -236,7 +236,6 @@ static bool readCachedModule(llvm::MemoryBuffer *in,
     auto diagSeverity =
         static_cast<CodeCompletionDiagnosticSeverity>(*cursor++);
     auto isSystem = static_cast<bool>(*cursor++);
-    auto isAsync = static_cast<bool>(*cursor++);
```
Comment (ahoppen) [inline lib/IDE/CodeCompletionCache.cpp:239]: Did you increment the completion cache version number?

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**